packages/cli/generators/controller/templates/src/controllers/controller-rest-template.ts.ejs

```diff
@@ -8,14 +8,13 @@ import {
   del,
   requestBody
 } from '@loopback/openapi-v3';
-import {inject} from '@loopback/context';
 import {<%= modelName %>} from '../models';
 import {<%= repositoryName %>} from '../repositories';
 
 export class <%= name %>Controller {
 
   constructor(
-    @inject('repositories.<%= repositoryName %>')
+    @repository(<%= repositoryName %>.name)
```

Comment:
-1

Any reason we can't just do the `.name` in our implementation of the `@repository` decorator rather than have the user add it? Poor UX imo and requires people to understand what is happening here. 

We should be able to resolve a Dependency given a Class imo. `@repository(UserRepository)`. 

Out of scope of this PR but I think this can apply in general to `@inject` as follows `@inject(UserController)` should be enough for us to be able to resolve this type. IIRC Angular's DI framework is capable of resolving an instance given a Class.

Reply:
Although I agree that the decorator should be able to resolve from just the class, it's outside of the scope for tasks in #744. I'll agree to change `@repository` and `@inject` arguments so that it can resolve the dependencies just from the class, but only if everyone unanimously agrees that this is our approach going forward.

Reply:
It makes sense to use the class for DI. However, I think we should create an issue for it and address it separately to get this patch landed for its intended purpose.

Reply:
+1 I originally intended to get miroslav's feedback to TV's proposal, but since that might not be coming soon I think I'd like to go ahead with the original plan

Reply:
+1 for allowing `@repository(UserRepository)`, +1 for leaving it out of scope of #744 and creating a new (follow-up) issue.